realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -428,6 +418,11 @@ private void enableCollectionSnapshot() {
     }
 
     void disableCollectionSnapshot() {
+        if (isInTransaction()) {
+            // This should never happen.
+            throw new IllegalStateException( "Collection cannot be reattached if the Realm is in transaction." +
+                    " Please remember to commit or cancel transaction before finishing the current event loop.");
```

Comment:
@Zhuinden Yes, although this check normally won't fail. And isn't across loops transaction a bad practise? If the transaction is not finished in one loop: 1) the Realm won't be updated since it is in transaction. 2) it would be easy to make mistakes in the code, like never commit the transaction.